src/execution/error.rs

```diff
@@ -124,6 +151,10 @@ impl<R> ResultWithExecutionError<R> for Result<R, ExecutionError> {
     {
         self.map_err(|e| match e {
             cancelled @ ExecutionError::Cancelled(_) => cancelled,
+            in_other_context @ ExecutionError::InContext(Context::Other(_), _) => {
+                ExecutionError::InContext(with_context(), Box::new(in_other_context))
+            }
+            in_stmt_context @ ExecutionError::InContext(_, _) => in_stmt_context,
```

Comment:
We only want to report the innermost statement (in phase one), or the originating statement (in phase two). Therefore we ignore any statement context if another statement context is already attached to the error. Other contexts (which are strings) do not prevent attaching a statement context.